packages/frontend-v2/hooks/useLocalStorage.ts

```diff
@@ -0,0 +1,35 @@
+import { useState, Dispatch, SetStateAction } from "react";
+
+const isWindow = typeof window !== "undefined";
+
+export function useLocalStorage<T>(
+  key: string,
+  initialValue: T
+): [T, Dispatch<SetStateAction<T>>] {
+  const [storedValue, setStoredValue] = useState<T>(() => {
+    try {
+      const item = isWindow && window.localStorage.getItem(key);
+      return item ? JSON.parse(item) : initialValue;
+    } catch (error) {
+      console.error(error);
+      return initialValue;
+    }
+  });
+
+  const setValue = (value: T | ((value: T) => T)) => {
+    try {
+      const valueToStore =
+        value instanceof Function ? value(storedValue) : value;
+      setStoredValue(valueToStore);
+
+      isWindow &&
+      window.localStorage.setItem(key, JSON.stringify(valueToStore));
+    } catch (error) {
+      console.error(error);
+    }
+  };
+
+
```

Comment:
I think Brandon did try to add in the remove value function last week during standup, but there were some type mismatches. Later decided not to add it because the type mismatch was kind of annoying to fix and didn't seem worth to spend more time on since we might not need it for now. But could a comment on that